src/PostgREST/Query.hs

```diff
@@ -278,3 +281,15 @@ runPreReq conf = lift $ traverse_ (SQL.statement mempty . stmt) (configDbPreRequ
       ("select " <> fromQi req <> "()")
       HD.noResult
       (configDbPreparedStatements conf)
+
+-- | Sets Timezone from prefer header
+setTimezone :: TimezoneNames -> Preferences -> DbHandler ()
+setTimezone allowedTimezones preferences =
+  lift $
+    when (isJust $ L.find (== timezone) allowedTimezones)
+    $ do
+        SQL.sql $ "set local time zone '" <> T.encodeUtf8 timezone <> "'"
+        SQL.sql $ "select set_config('timezone', '" <> T.encodeUtf8 timezone <> "', true)"
```

Comment:
Isn't `set_config(timezone,...)` the only setting required here? As Steve suggested, this should be set alongside the other configurations, here:

https://github.com/PostgREST/postgrest/blob/b2352271194cd0051905fdd73232c6b23b29f825/src/PostgREST/Query.hs#L239-L247